### PR TITLE
Fixes #15333 - Moved puppet environment field to helper

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -100,6 +100,60 @@ def realm_field(f, can_override = false, override = false)
             ).html_safe
   end
 
+  def host_puppet_environment_field(f, select_options = {}, html_options = {})
```
Comment (shlomizadok) [inline app/helpers/hosts_and_hostgroups_helper.rb]: Maybe a more meaningful variable name than `f`? (form..)

**Comment by shlomizadok** [comment]
Cool. Had one small nit

File: app/helpers/hosts_and_hostgroups_helper.rb (hunk 1 above)
Comment (ShimShtein) [inline app/helpers/hosts_and_hostgroups_helper.rb]: It's kind of convention - most of methods use it, just scroll up :)
If you still want to change it, I'll do it.

Comment (shlomizadok) [inline app/helpers/hosts_and_hostgroups_helper.rb]: Not for me to decide whether to change or not... 
(but maybe time to change that convention - `f` means nothing here)

**Comment by ShimShtein** [comment]
Changed `f` parameter to `form`

**Comment by ares** [comment]
@ShimShtein there are 4 test failures

```
ActionView::Template::Error: undefined local variable or method `f' for #<#<Class:0x00000043b848c0>:0x0000000f1e9628>
```

https://github.com/theforeman/foreman/pull/3578/files#diff-49521b3e6ed59e866e058b10bca549fdR127

**Comment by ShimShtein** [comment]
Updated, sorry about that.

**Comment by dLobatog** [comment]
Merged as 41ef1c8b02b5e2c6fbb34b652c0156d58d9b8d8a, thanks @ShimShtein!

**Pull request closed without merging**